core/src/com/unciv/logic/automation/unit/UnitAutomation.kt

```diff
@@ -127,6 +129,8 @@ object UnitAutomation {
     }
 
     internal fun tryUpgradeUnit(unit: MapUnit): Boolean {
+        val isPlayer = unit.civ.isCurrentPlayer()
```

Comment:
This is not the right check, this could prevent AI as well and is less indicative of what we're actually trying to avoid :)
What we want it to check unit.civ.isHuman()